pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -195,66 +204,203 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
 
-        if (multiSzValue) {
-          // Registry value can be a single-string or even a DWORD and parseMultiString will handle it.
-          const arrayValue = nativeReg.parseMultiString(multiSzValue as nativeReg.Value);
+    if (!innerKey) {
+      continue;
+    }
+    try {
+      regValue = readRegistryUsingSchema(schemaVal, innerKey, pathParts.concat([schemaKey]));
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[schemaKey] = regValue;
+    }
+  }
+  for (const [schemaKey, registryKey] of specialObjectKeys) {
+    const innerKey = nativeReg.openKey(regKey, registryKey, nativeReg.Access.READ);
 
-          regValue = arrayValue.length ? arrayValue : null;
-        }
-      }
+    if (innerKey === null) {
+      console.error(`No value for registry object ${ pathParts.join('\\') }\\${ registryKey }`);
+      continue;
+    }
+    try {
+      regValue = readRegistryObject(innerKey, pathParts.concat([schemaKey]), true);
+    } catch (err: any) {
+      console.error(`Error getting registry object for ${ pathParts.join('\\') }\\${ registryKey }: `, err);
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[schemaKey] = regValue;
+    }
+  }
+  const unknownValueNames: string[] = [];
+  const registryValueNames = nativeReg.enumValueNames(regKey);
+  const registryValueNamesFixedCase = registryValueNames.map(k => fixProfileKeyCase(k, schemaKeys));
+
+  registryValueNamesFixedCase.forEach((key, index) => {
+    const originalName = registryValueNames[index];
+
+    if (key === '') {
+      unknownValueNames.push(originalName);
     } else {
-      regValue = nativeReg.queryValue(regKey, k);
-      if (typeof schemaVal === 'boolean') {
-        if (typeof regValue === 'number') {
-          regValue = regValue !== 0;
-        } else {
-          console.debug(`Deployment Profile expected boolean value for key ${ k }`);
-          regValue = false;
-        }
+      regValue = readRegistryValue(schemaObj[key], regKey, pathParts, originalName);
+      if (regValue !== null) {
+        newObject ??= {};
+        (newObject as Record<string, any>)[key] = regValue;
       }
     }
-    if (regValue !== null) {
-      newObject ??= {};
-      (newObject as Record<string, any>)[k] = regValue;
+  });
+  if (unknownValueNames.length > 0) {
+    console.error(`Unrecognized value names in registry at ${ pathParts.join('\\') }: [${ unknownValueNames.join(', ') }]`);
+  }
+
+  return newObject;
+}
+
+function fixProfileKeyCase(key: string, schemaKeys: string[]): string {
+  const idx = schemaKeys.findIndex(val => isEquivalentIgnoreCase(key, val));
+
+  return idx === -1 ? '' : schemaKeys[idx];
+}
+
+function readRegistryObject(regKey: nativeReg.HKEY, pathParts: string[], isSpecialObject = false) {
```

Comment:
I think the naming of `isSpecialObject` is very confusing. My understanding (from `readRegistryUsingSchema`) is that it means it's reading an object that doesn't have a default; however, this seems to do some kind of convert to boolean, and `readRegistryValue` seems to do something about strings too? Could we actually document that (as a comment or something) what that's meant to do?

Reply:
No, it's reading a registry object that doesn't have a corresponding schema. I prefer 'user-defined object' to 'schemaless object', so I'll go with that.